src/core/worker.js

```diff
@@ -623,6 +508,16 @@ var WorkerMessageHandler = {
             }
 
             if (pdfManager) {
+              // There may be a chance that pdfManager is not initialized
+              // for some previous runs of this(readchunk) block of code.
+              // Be sure to send all cached chunks, if any, to chunked_stream
+              // via pdf_manager.
+              if (cachedChunks.length !== 0) {
+                for (let i = 0; i < cachedChunks.length; i++) {
+                  pdfManager.sendProgressiveData(cachedChunks[i]);
+                }
+                cachedChunks = [];
+              }
```

Comment:
Can we move that to the https://github.com/mozilla/pdf.js/pull/8617/files#diff-780c33139a29f60c0cc45b80fb05f900R443  (just after we created pdfManager)?